src/commands/signzone.rs

```diff
@@ -0,0 +1,1397 @@
+                    zonemd.push(parse_os_with("-z", &val, SignZone::parse_zonemd_tuple)?);
+                }
+                Arg::Short('Z') => {
+                    allow_zonemd_without_signing = true;
+                }
+                Arg::Short('v') => {
+                    let version = clap::crate_version!();
+                    println!("zone signer version {version} (dnst version {version})");
+                    std::process::exit(0);
+                }
+                Arg::Short('n') => {
+                    use_nsec3 = true;
+                }
+                Arg::Short('a') => {
+                    let val = parser.value()?;
+                    algorithm = parse_os_with("-a", &val, Nsec3Hash::parse_nsec3_alg)?;
+                }
+                Arg::Short('t') => {
+                    let val = parser.value()?;
+                    iterations = parse_os("-t", &val)?;
+                }
+                Arg::Short('s') => {
+                    let val = parser.value()?;
+                    salt = parse_os("-s", &val)?;
+                }
+                Arg::Short('p') => {
+                    nsec3_opt_out = true;
+                }
+                Arg::Value(val) => {
+                    if zonefile.is_none() {
+                        zonefile = Some(parse_os("zonefile", &val)?);
+                    } else {
+                        key_paths.push(parse_os("key", &val)?);
+                    }
+                }
+                Arg::Short(x) => return Err(format!("Invalid short option: -{x}").into()),
+                Arg::Long(x) => {
+                    return Err(format!("Long options are not supported, but `--{x}` given").into())
+                }
+            }
+        }
+
+        let Some(zonefile_path) = zonefile else {
+            return Err("Missing zonefile argument".into());
+        };
+
+        if let Some(out_file) = &out_file {
+            if out_file.as_os_str() == "-" {
+                extra_comments = false;
+            }
+        }
+
+        // Logically this should also check that zonemd flags are given, but
+        // ldns-signzone just copies the unsigned zone (without comments) when
+        // using only -Z (without -z).
+        if key_paths.is_empty() && !allow_zonemd_without_signing {
+            return Err("Missing key argument".into());
+        };
+
+        Ok(Self {
+            extra_comments,
+            do_not_add_keys_to_zone,
+            expiration,
+            out_file,
+            inception,
+            origin,
+            set_soa_serial_to_epoch_time,
+            zonemd,
+            allow_zonemd_without_signing,
+            use_nsec3,
+            algorithm,
+            iterations,
+            salt,
+            nsec3_opt_out_flags_only: true,
+            // TODO: is mixed up with flags_only? The commend on clap args says
+            // nsec3_opt_out is not originally supported, but flags_only is
+            nsec3_opt_out,
```

Comment:
The comment on the clap argument says that the `nsec3_opt_out` is not originally supported by ldns-signzone, but `nsec3_opt_out_flags_only` is. Is this a mix up in the ldns parsing then?